[16.0][ADD] account_multi_foreign_vat

ArnauCForgeFlow opened the pull request.

ArnauCForgeFlow committed:
[ADD] account_multi_foreign_vat

ArnauCForgeFlow committed:
[IMP] account_multi_foreign_vat: allows multiple fp from same country

LoisRForgeFlow commented on account_multi_foreign_vat/models/partner.py:
```diff
@@ -0,0 +1,30 @@
+
+class AccountFiscalPosition(models.Model):
+
+    _inherit = "account.fiscal.position"
```
You are inheriting twice account.fiscal.position in tow different files. Can you move this extension to the file with the name of the model?

ArnauCForgeFlow replied:
Sure, my bad

LoisRForgeFlow commented on account_multi_foreign_vat/models/account_fiscal_position.py:
```diff
@@ -0,0 +1,39 @@
+                            "number does not match the one set on this fiscal position."
+                        )
+                    ) from e
+                if not checked_country_code and not record.country_id:
+                    raise ValidationError(
+                        _("The foreign VAT number is not correct.")
+                    ) from e
```
Is this the only change done to the standard method? https://github.com/odoo/odoo/blob/7d678cc7f4c9e00d78d4120f703e90bd282c1a51/addons/base_vat/models/account_fiscal_position.py

ArnauCForgeFlow replied:
Yes

LoisRForgeFlow commented on account_multi_foreign_vat/models/partner.py:
```diff
@@ -0,0 +1,30 @@
+            try:
+                super()._validate_foreign_vat_country()
+            except Exception:
+                similar_fpos_domain = [
+                    ("foreign_vat", "!=", False),
+                    ("country_id", "=", record.country_id.id),
+                    ("company_id", "=", record.company_id.id),
+                    ("id", "!=", record.id),
+                ]
+                if record.state_ids:
+                    similar_fpos_domain.append(
+                        ("state_ids", "in", record.state_ids.ids)
+                    )
+
+                similar_fpos_count = self.env["account.fiscal.position"].search_count(
+                    similar_fpos_domain
+                )
+                if similar_fpos_count:
+                    pass
```
at the end this code is just skipping the full check, isn't it? Why dont you just return pass?

ArnauCForgeFlow replied:
Yes, you are right

LoisRForgeFlow commented on account_multi_foreign_vat/README.rst:
```diff
@@ -0,0 +1,41 @@
+======================================
+Account Multi Foreign VAT
```
README files in OCA are autogenerated from fragments, see https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

ArnauCForgeFlow replied:
Okay, I didn't know

LoisRForgeFlow reviewed (commented):
Some comments

ArnauCForgeFlow committed:
[IMP] account_multi_foreign_vat: fixed some review comments

LoisRForgeFlow committed:
[FIX] account_multi_foreign_vat: consider new cases also when creating entries

- if the impacted countries are part of the country group of the fiscal position
  we do not raise an error.
- if the fiscal position does not have country or country group specificed, we
  do not raise an error.

ArnauCForgeFlow committed:
[IMP][ADD] account_multi_foreign_vat: added tests, and improved readme

ArnauCForgeFlow committed:
[IMP] account_multi_foreign_vat

dreispt commented:
Are requested changes done @LoisRForgeFlow ?

JordiMForgeFlow committed:
[IMP] account_multi_foreign_vat: allow selecting different country taxes

JordiMForgeFlow commented:
@dreispt could we reopen this one?
